### PR TITLE
[API BREAK] Make flags more explicit, add runtime checks.

### PR DESCRIPTION
**Pull request by rustyrussell**
Fewer zeroes, plus check bit to make sure it value is valid.

Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

### CODE REVIEW

**Comment by jgarzik** [comment]
ut ACK

File: include/secp256k1.h [hunk 1]
```diff
@@ -148,12 +148,21 @@ typedef int (*secp256k1_nonce_function)(
...
-# define SECP256K1_CONTEXT_VERIFY (1 << 0)
-# define SECP256K1_CONTEXT_SIGN   (1 << 1)
-
+# define SECP256K1_CONTEXT_VERIFY (SECP256K1_CONTEXT_VERIFY_BIT|SECP256K1_CONTEXT_CHECK_BIT)
+# define SECP256K1_CONTEXT_SIGN   (SECP256K1_CONTEXT_SIGN_BIT|SECP256K1_CONTEXT_CHECK_BIT)
+# define SECP256K1_CONTEXT_NONE   (SECP256K1_CONTEXT_CHECK_BIT)
+
+#  define SECP256K1_CONTEXT_VERIFY_BIT     (1 << 2)    
+#  define SECP256K1_CONTEXT_SIGN_BIT       (2 << 2)    
+#  define SECP256K1_CONTEXT_CHECK_BIT      (1)
```
Comment (apoelstra) [inline include/secp256k1.h]: I'd prefer VERIFY and SIGN to be (1 << 0) and (1 << 1); COMPRESSED should also be (1 << 0), and in general the actual flags should be written as (1 << n) for increasing n.

The CHECK_BITs would then need to be something high, maybe (1 << 10), (1 << 11), etc, which would give us room for ten flags in each of five different flag categories, while still fitting into a 16-bit int.

**Comment by apoelstra** [comment]
Can you add [API BREAK] to the title of this PR?

Also concept ACK.

**Comment by sipa** [comment]
Superseded by #339.

**Pull request closed without merging**